src/ngraph/runtime/gpu/gpu_backend.cpp

```diff
@@ -114,6 +115,10 @@ shared_ptr<runtime::Tensor>
 shared_ptr<runtime::Tensor> runtime::gpu::GPU_Backend::create_tensor(
     const element::Type& element_type, const Shape& shape, void* memory_pointer)
 {
+    if (memory_pointer != nullptr && !is_device_pointer(memory_pointer))
+    {
+        throw ngraph_error("The pointer passed to create_tensor is not a device pointer.");
```

Comment:
We can create a device tensor here and copy the data over, but that might be a separate PR.

Reply:
ya, I was also think in that way. one concern was user might assume the pointer will be used by backend. which is not true. it's likely user will miss the data sync.

Reply:
We can handle that in another PR.